### PR TITLE
feat: UV

### PR DESCRIPTION
**Pull request by Sung96kim**
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] Jira story/task is put into PR title
- [ ] Code has been self-reviewed
- [ ] Hard-to-understand areas have been commented
- [ ] Documentation has been updated
- [ ] My changes generate no new warnings
- [ ] Unit tests have been added for base functionality and known edge cases
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by jacobmanderson — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -37,6 +37,7 @@ source = "uv-dynamic-versioning"
 [tool.uv-dynamic-versioning]
 fallback-version = "0.0.0"
 vcs = "git"
+pattern = "^(?P<base>\\d+(?:\\.\\d+)*)(?:-(?P<stage>[a-zA-Z]+)(?P<revision>\\d+)?)?"
 
 [tool.uv]
 cache-keys = [{ file = "pyproject.toml" }, { git = { commit = true, tags = true } }]
```
Comment (thearchitector) [inline pyproject.toml]: what dis do?

**Review by jacobmanderson — APPROVED** [review]
